include/dca/phys/models/analytic_hamiltonians/twoband_Cu.hpp

```diff
@@ -0,0 +1,284 @@
+
+  static std::vector<std::pair<std::pair<int, int>, std::pair<int, int>>> get_orbital_permutations();
+
+  // Initializes the interaction Hamiltonian in real space.
+  template <typename BandDmn, typename SpinDmn, typename RDmn, typename parameters_type>
+  static void initialize_H_interaction(
+      func::function<double, func::dmn_variadic<func::dmn_variadic<BandDmn, SpinDmn>,
+                                                func::dmn_variadic<BandDmn, SpinDmn>, RDmn>>& H_interaction,
+      const parameters_type& parameters);
+
+  template <class HType, class Parameters>
+  static void initializeNonDensityInteraction(HType& interaction, const Parameters& pars);
+
+  template <class domain>
+  static void initialize_H_symmetry(func::function<int, domain>& H_symmetry);
+
+  // Initializes the tight-binding (non-interacting) part of the momentum space Hamiltonian.
+  // Preconditions: The elements of KDmn are two-dimensional (access through index 0 and 1).
+  template <typename ParametersType, typename ScalarType, typename BandDmn, typename SpinDmn, typename KDmn>
+  static void initialize_H_0(
+      const ParametersType& parameters,
+      func::function<ScalarType, func::dmn_variadic<func::dmn_variadic<BandDmn, SpinDmn>,
+                                                    func::dmn_variadic<BandDmn, SpinDmn>, KDmn>>& H_0);
+};
+
+template <typename PointGroup>
+int TwobandCu<PointGroup>::transformationSign(int b1, int b2, int s) {
+  if (!std::is_same<PointGroup, domains::D4>::value)
+    return 1;
+
+  if (b1 == b2)
+    return 1;
+
+  //  using List = typename PointGroup::PointGroup_list;
+  using dca::util::IndexOf;
+
+  constexpr bool symmetrize_off_diagonal = true;
+
+  if (symmetrize_off_diagonal) {
+    //    const bool is_odd_rotation = IndexOf<domains::Cn_2D<1, 4>, List>::value == s ||
+    //                                 IndexOf<domains::Cn_2D<3, 4>, List>::value == s;
+    // TODO: generalize.
+    const bool is_odd_rotation = (s % 2) == 1;
+
+    return is_odd_rotation ? -1 : 1;
+  }
+
+  else {
+    // TODO: generalize.
+    const bool is_identity = s == 0;
+    return is_identity ? 1 : 0;
+  }
+}
+
+template <typename PointGroup>
+double* TwobandCu<PointGroup>::initialize_r_DCA_basis() {
+  static std::array<double, 4> r_DCA{1, 0, 0, 1};
+  return r_DCA.data();
+}
+
+template <typename PointGroup>
+double* TwobandCu<PointGroup>::initialize_k_DCA_basis() {
+  static std::array<double, 4> k_DCA{2 * M_PI, 0, 0, 2 * M_PI};
+  return k_DCA.data();
+}
+
+template <typename PointGroup>
+double* TwobandCu<PointGroup>::initialize_r_LDA_basis() {
+  static std::array<double, 4> basis{1, 0, 0, 1};
+  return basis.data();
+}
+
+template <typename PointGroup>
+double* TwobandCu<PointGroup>::initialize_k_LDA_basis() {
+  static std::array<double, 4> basis{2 * M_PI, 0, 0, 2 * M_PI};
+  return basis.data();
+}
+
+template <typename PointGroup>
+std::vector<int> TwobandCu<PointGroup>::get_flavors() {

```

Comment:
none of these should be get_ they are not simple getters or setters in fact they wrap singleton like vectors.

Reply:
updated the names in ad2485498f0d2e09b41105fb887dedbeb9e90280